cmd/hendlers/hendlers.go

```diff
@@ -0,0 +1,85 @@
+package hendlers
+
+import (
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type Subj struct {
+	//sync.Mutex
+	ID      string `json:"id"`
+	LongURL string `json:"LongURL"`
+}
+
+type JSON struct {
+	sync.Mutex
+	Obj []Subj
+}
+
+var sett JSON
+
+func BodyHandler(w http.ResponseWriter, r *http.Request) {
+	sett.Mutex.Lock()
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(400)
+	}
+
+	defer r.Body.Close()
+
+	str := string(b)
+	sitr, err := url.PathUnescape(str)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	id := strconv.Itoa(rand.Int())
+	StrURL := "http://localhost:8080/" + id
+
+	w.WriteHeader(http.StatusCreated)
+
+	var settings JSON
+
+	newURL := Subj{
+		ID:      id,
+		LongURL: sitr,
+	}
+
+	sett.Obj = append(settings.Obj, newURL)
+
+	w.Write([]byte(StrURL))
+
+	sett.Mutex.Unlock()
+
+}
+
+func GetHandler(w http.ResponseWriter, r *http.Request) {
+	sett.Mutex.Lock()
+	//q := r.URL.String()
+	q := chi.URLParam(r, "id")
+	fmt.Println(sett.Obj)
+	for _, v := range sett.Obj {
+
+		if q == v.ID {
+
+			Location := strings.TrimSpace(v.LongURL)
+			w.Header().Set("Location", v.LongURL)
+			fmt.Println(Location)
+			if Location != "" {
+				http.Redirect(w, r, Location, http.StatusTemporaryRedirect)
+			} else {
+				w.WriteHeader(400)
```

Comment:
```suggestion
				w.WriteHeader(400)
				return
```
иначе рискуем навыдавать кучу статусов в одном ответе.